### PR TITLE
Add bounds checking for scan; fix formatting of scan warnings

### PR DESCRIPTION
**Pull request by liu15**

### CODE REVIEW

**Review by robinson96 — APPROVED** [review]
LGTM

File: src/care/scan_impl.h [hunk 1]
```diff
@@ -43,21 +44,49 @@ void exclusive_scan(chai::ManagedArray<T> data, //!< [in/out] Input data (output
...
-            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false and size > 1, data cannot be nullptr.");
+            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false and size > 1, data cannot be nullptr.\n");
             return;
          }
       }
 
+#if !defined(CHAI_DISABLE_RM)
+      // Bounds checking is only available with the resource manager because the ManagedArray::size() is not
+      // reliable if it is cast to a different template type.
+      const chai::PointerRecord* dataRecord = chai::ArrayManager::getInstance()->getPointerRecord((void *)data.cdata());
```
Comment (liu15) [inline src/care/scan_impl.h:65]: should I be explicitly including the umpire headers needed for the umpire calls?

Comment (robinson96) [reply src/care/scan_impl.h:65]: do you mean the chai calls? I could see an include of chai/PointerRecord.hpp (or whatever it is) to be explicit.

Comment (liu15) [reply src/care/scan_impl.h:65]: Oops, I have to update the PR.... I meant the backtrace from umpire.

**Review by adayton1 — APPROVED** [review]
LGTM

**Pull request merged**